matchzoo/layers/match_layer.py

```diff
@@ -0,0 +1,129 @@
+"""An implementation of Match Layer."""
+from __future__ import absolute_import
+
+import typing
+from keras import backend as K
+from keras.engine import Layer
+
+
+class MatchLayer(Layer):
```

Comment:
also it's a bit ambiguous, since any layer could be a Matching Layer, for example, MultiPerspectiveMatching is more descriptive.